src/mutation/__tests__/RelayMutationQuery-test.js

```diff
@@ -43,6 +43,11 @@ describe('RelayMutationQuery', () => {
       node => !node.getSchemaName || node.getSchemaName() !== 'source';
     return filterRelayQuery(RelayTestUtils.getNode(...args), filterCallback);
   }
+  function getMutationWithoutClientId(mutation) {
```

Comment:
Maybe it makes more sense to fetch the clientMutationId if the clientMutationId input field is non-null so that relay knows whether it needs to provide the clientMutationId in the input.  That would avoid any need for breaking changes.